cache/in-memory/src/builder.rs

```diff
@@ -27,6 +28,43 @@ impl InMemoryCacheBuilder {
         self
     }
 
+    /// Sets the list of rules associated with the cache.
+    ///
+    /// Defaults to none.
+    ///
+    /// # Examples
+    ///
+    /// Add a rule to the cache:
+    ///
+    /// ```
+    /// use twilight_cache_inmemory::{config::{Entity, Rule}, InMemoryCache};
+    ///
+    /// #[derive(Clone, Debug, Eq, PartialEq)]
+    /// pub struct HyphenatedRoleFilter;
+    ///
+    /// impl Rule for HyphenatedRoleFilter {
+    ///     fn accept(&self, cache: &InMemoryCache, entity: &Entity) -> bool {
+    ///         let role = if let Entity::Role(role) = entity {
+    ///             role
+    ///         } else {
+    ///             return true;
+    ///         };
+    ///
+    ///         role.name.starts_with('-')
+    ///     }
+    /// }
+    ///
+    /// let rules = Vec::from([Box::new(HyphenatedRoleFilter) as Box<dyn Rule>]);
+    ///
+    /// let cache = InMemoryCache::builder().rules(rules).build();
+    /// assert_eq!(1, cache.config().rules().len());
+    /// ```
+    pub fn rules(mut self, rules: Vec<Box<dyn Rule>>) -> Self {
```

Comment:
May I also request two additional functions, a `rule` function to allow adding a single rule, and an `add_rules` function to add additional rules without overriding the already set rules